### ISSUE TITLE
Adding links and setting directory

### ISSUE THREAD

**Issue by devocalypse** [Cr1TiKa7/My.Jdownloader.Api#5]
In a scenario where we want to add links and change the download location:
```csharp
Device.LinkgrabberV2.AddLinks(
                new My.JDownloader.Api.Models.LinkgrabberV2.Request.AddLinkRequest
                {
                    Links = url,
                    DestinationFolder = destination,  //ignored always
                    PackageName = package,  //ignored for folders
                    AutoStart = autostart
                });
```
When adding a single file link PackageName is correct, DestinationFolder is ignored.
When adding a folder link (with multiple files), Both PackageName and DestinationFolder are ignored.
This must be jdownloaders fault for handling it poorly.

The workaround would be to AddLinks > Wait for it to finish > Query packages > SetDownloadDirectory on package IDs
However waiting seems broken in this case:
```csharp
        public bool IsCollecting()
        {
            if (ApiHandler.CallAction<object>(Device, "/linkgrabberv2/isCollection", null, JDownloaderHandler.LoginObject) == null)
            {
                return false;
            }
            return true;
        }
```
The method name could have changed in the api since there it states the method should be
**isCollecting** instead of **isCollection**
https://my.jdownloader.org/developers/index.html#tag_255

**Comment by Viperinius** [comment]
The first issue can be solved if you use the `overwritePackagizerRules` parameter. If this is passed as `true`, both destination and package name seem to work.

**Comment by devocalypse** [comment]
> The first issue can be solved if you use the `overwritePackagizerRules` parameter. If this is passed as `true`, both destination and package name seem to work.

Thanks for the suggestion.

Adding this to AddLinkRequest.cs class:
`[JsonProperty(PropertyName = "overwritePackagizerRules")]
        public bool OverridePackagizer { get; } = true;`

Actually did the trick.

**Comment by Cr1TiKa7** [comment]
Thank you guys for reporting! 
The isCollecting endpoint propably never changed and it is very likely my fault but i'm not sure anymore since it is already 3 years ago that I've started this project. But i've renamed it to isCollecting.

Also added the OverwritePackagizerRules property to the AddLinkRequest object.

Both fixes included in the new version 2.4.4
